src/u2f.c

```diff
@@ -101,6 +106,82 @@ typedef struct __attribute__((__packed__)) {
     uint8_t challenge[U2F_NONCE_LENGTH];
 } U2F_AUTHENTICATE_SIG_STR;
 
+typedef enum {
+    U2F_STATE_IDLE,
+    U2F_STATE_AUTH,
+    U2F_STATE_REGISTER,
+} u2f_state_t;
+
+/**
+ * Holds state data for the U2F_STATE_AUTH workflow.
+ */
+typedef struct {
+    /** Part of the auth we're currently processing. */
+    enum {
+        U2F_STATE_AUTH_REQUESTED,
```

Comment:
please document each of those as clearly as possible

Reply:
Old code, was refactored.